common/src/test/scala/uk/ac/wellcome/message/MessageWorkerTest.scala

```diff
@@ -0,0 +1,154 @@
+package uk.ac.wellcome.message
+
+import org.mockito.Matchers.{any, anyDouble, anyString, contains, matches}
+import org.mockito.Mockito.{never, times, verify, when}
+import org.scalatest.FunSpec
+import org.scalatest.concurrent.Eventually
+import org.scalatest.mockito.MockitoSugar
+import uk.ac.wellcome.utils.JsonUtil._
+import uk.ac.wellcome.test.fixtures._
+import uk.ac.wellcome.s3.S3Uri
+import uk.ac.wellcome.test.utils.ExtendedPatience
+import uk.ac.wellcome.sns.NotificationMessage
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.collection.JavaConversions._
+
+class MessageWorkerTest
+    extends FunSpec
+    with MockitoSugar
+    with Eventually
+    with ExtendedPatience
+    with Messaging {
+
+  it("processes messages") {
+    withMessageWorkerFixtures {
+      case (_, queue, metrics, bucket, worker) =>
+        val key = "message-key"
+
+        val exampleObject = ExampleObject("some value")
+        val exampleObjectJson = toJson(exampleObject).get
+
+        val examplePointer = MessagePointer(S3Uri(bucket.name, key))
+
+        val exampleNotification = NotificationMessage(
+          MessageId = "MessageId",
+          TopicArn = "TopicArn",
+          Subject = "Subject",
+          Message = toJson(examplePointer).get
+        )
+
+        s3Client.putObject(bucket.name, key, exampleObjectJson)
+
+        sqsClient.sendMessage(
+          queue.url,
+          toJson(exampleNotification).get
+        )
+
+        eventually {
+          verify(
```

Comment:
Sending metrics s not the primary feature of `MessageWorker`. The main thing is that it calls the `processMessage` function. Can we make a different assertion that reflects that?

Reply:
See `SQSReaderTest` for an example of this.